### ISSUE TITLE
Clear storage (userred, useryellow, usergreen, blockedOrigins, etc) when addon is uninstalled?

### ISSUE THREAD

**Issue by diracdeltas** [EFForg/privacybadgerfirefox#10]
This doesn't happen automatically, but it seems like it should. On the other hand, maybe users who care about this for privacy reasons would just delete their Firefox profile.

**Comment by cooperq** [comment]
via @Hainish 
>Perhaps a desirable solution to this is to clear the cache upon
installation? Or is persistence a desired property? Either way, it should
behave consistently across FF/Chrome.

**Comment by cooperq** [comment]
We had a clear data button, but I removed it. I guess I should probably add it again.

**Comment by MHWallace** [comment]
I vote for adding a clear data button into PB Options. As a user I would like to be able to reset back to settings that PB has when first installed. (clear PBadger's local stores and reset cookie permissions set by PBadger.) Not knowing the inner workings of PB I'm not sure exactly what is possible but my idea is to reset to settings as they were "out of the box".

**Comment by dvdbane** [comment]
You may already be working on adding Clear all data button but is it possible to add another button to delete all except user custom setting?

**Comment by cooperq** [comment]
I would be reluctant to add such a button.

**Comment by oemb1905** [comment]
Please re-add the clear settings button.

**Comment by MHWallace** [comment]
Agreed about Cooperq's reluctance on "delete all except user custom settings" button. That is opposite what I am looking for. Refining Oemb1905's thought to add a "clear user custom settings" button that preserves default settings. I don't want to clear all settings, just the custom settings I've set over time. Reiterating my initial idea is reset to settings as they were "out of the box". I do this with NoScript settings using their "Reset" button.

**Comment by oemb1905** [comment]
Forgive me for being brief.  Yes, I am looking for a Reset to Default option similar to the No Script extension; just as MHWallace describes.  Regards, oemb1905
